tools/api-markdown-documenter/src/api-item-transforms/configuration/Hierarchy.ts

```diff
@@ -0,0 +1,346 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
```

Comment:
This module is more or less the core of these changes. Control over the output suite hierarchy is unified here (in place of the old `documentBoundaries` and `hierarchyBoundaries` properties on `DocumentationSuiteConfiguration`.

The core goal of these changes is to allow configuring where an API item's document is placed relative to its directory (inside or outside), for items that yield folder hierarchy.